nautobot/extras/datasources/git.py

```diff
@@ -816,24 +826,27 @@ def refresh_git_jobs(repository_record, job_result, delete=False):
     """Callback function for GitRepository updates - refresh all Job records managed by this repository."""
     installed_jobs = []
     if "extras.job" in repository_record.provided_contents and not delete:
-        jobs_path = os.path.join(repository_record.filesystem_path, "jobs")
-        if os.path.isdir(jobs_path):
-            for job_info in jobs_in_directory(jobs_path, report_errors=True):
-                if job_info.error is not None:
-                    job_result.log(
-                        message=f"Error in loading Jobs from `{job_info.module_name}`: `{job_info.error}`",
-                        grouping="jobs",
-                        level_choice=LogLevelChoices.LOG_ERROR,
-                        logger=logger,
-                    )
-                    continue
+        from nautobot.core.celery import app
 
-                job_model, created = refresh_job_model_from_job_class(
-                    Job,
-                    JobSourceChoices.SOURCE_GIT,
-                    job_info.job_class,
-                    git_repository=repository_record,
-                )
+        found_jobs = False
+        try:
+            if settings.GIT_ROOT not in sys.path:
+                sys.path.append(settings.GIT_ROOT)
+
+            module_name = f"{repository_record.slug}"
+            # Unload any previous version of this module if present
+            if module_name in sys.modules:
+                del sys.modules[module_name]
+            module_name += ".jobs"
+            if module_name in sys.modules:
+                del sys.modules[module_name]
+            app.loader.import_task_module(module_name)
```

Comment:
This should be extracted to a common helper function matching the logic in `nautobot.core.celery.control.refresh_git_repository` or possibly refactored to simply use that function instead.